src/domain/books/BooksPage.tsx

```diff
@@ -1,7 +1,30 @@
+import { useDispatch, useSelector } from "react-redux";
+import { BookCard } from "../../components/BookCard";
+import { useEffect } from "react";
+import "./styles.css";
+import { AppDispatch } from "../../store/store";
+import { booksSelector, getBooks } from "./booksSlice";
+
 export const BooksPage = (): JSX.Element => {
+  const dispatch: AppDispatch = useDispatch();
+  const Bookdata = useSelector(booksSelector);
+
+  useEffect(() => {
+    dispatch(getBooks());
+  }, [Bookdata]);
+
   return (
-    <div>
-      Books Page
+    <div className="book-page-container">
+      <>{console.log(Bookdata)}</>
+      {Bookdata.length === 0 ? (
+        <div>Loading...</div>
+      ) : (
```

Comment:
We get a loading state from the booksSelector, use it instead.

Reply:
Done